From f3fc43ec5681902ee14210714f344553d1221926 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Marcos=20Chavarr=C3=ADa=20Teijeiro?=
 <chavarria1991@gmail.com>
Date: Thu, 11 Sep 2014 12:08:18 +0200
Subject: yelp-view: Rename WKWebSettings to WKSettings and adapt properties

Remove enable-universal-access-from-file-uris that doesn't
exist any more and rename default-encoding to default-charset.
---
 libyelp/yelp-view.c | 19 ++++++++-----------
 1 file changed, 8 insertions(+), 11 deletions(-)

diff --git a/libyelp/yelp-view.c b/libyelp/yelp-view.c
index 50e4ed8..4067275 100644
--- a/libyelp/yelp-view.c
+++ b/libyelp/yelp-view.c
@@ -151,7 +151,7 @@ static gint signals[LAST_SIGNAL] = { 0 };
 G_DEFINE_TYPE (YelpView, yelp_view, WEBKIT_TYPE_WEB_VIEW)
 #define GET_PRIV(object)(G_TYPE_INSTANCE_GET_PRIVATE ((object), YELP_TYPE_VIEW, YelpViewPrivate))
 
-static WebKitWebSettings *websettings;
+static WebKitSettings *websettings;
 
 typedef struct _YelpActionEntry YelpActionEntry;
 struct _YelpActionEntry {
@@ -245,8 +245,6 @@ yelp_view_init (YelpView *view)
 {
     YelpViewPrivate *priv = GET_PRIV (view);
 
-    g_object_set (view, "settings", websettings, NULL);
-
     priv->cancellable = NULL;
 
     priv->prevstate = priv->state = YELP_VIEW_STATE_BLANK;
@@ -408,8 +406,11 @@ yelp_view_class_init (YelpViewClass *klass)
 
     nautilus_sendto = g_find_program_in_path ("nautilus-sendto");
 
-    websettings = webkit_web_settings_new ();
-    g_object_set (websettings, "enable-universal-access-from-file-uris", TRUE, NULL);
+    websettings = webkit_settings_new_with_settings (
+                    "default-charset", "utf-8",
+                    "enable-private-browsing", TRUE,
+                    NULL);
+
     g_signal_connect (settings,
                       "fonts-changed",
                       G_CALLBACK (settings_set_fonts),
@@ -591,7 +592,8 @@ yelp_view_set_property (GObject      *object,
 GtkWidget *
 yelp_view_new (void)
 {
-    return (GtkWidget *) g_object_new (YELP_TYPE_VIEW, NULL);
+    return GTK_WIDGET (g_object_new (YELP_TYPE_VIEW,
+                       "settings", websettings, NULL));
 }
 
 void
@@ -1928,11 +1930,6 @@ settings_set_fonts (YelpSettings *settings)
     gchar *family;
     gint size;
 
-    g_object_set (websettings,
-                  "default-encoding", "utf-8",
-                  "enable-private-browsing", TRUE,
-                  NULL);
-
     family = yelp_settings_get_font_family (settings,
                                             YELP_SETTINGS_FONT_VARIABLE);
     size = yelp_settings_get_font_size (settings,
-- 
cgit v0.12

